WcaOnRails/lib/relations.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Relations
+  def self.get_chain(wca_id1, wca_id2)
+    find_chain([[wca_id1]], [[wca_id2]])
+  end
+
+  def self.competitions_together(person1, person2)
+    Competition.find(person1.competition_ids & person2.competition_ids)
+  end
+
+  # Internal
+
+  def self.find_chain(left_chains, right_chains)
+    extended_chains_by_one_degree!(left_chains)
+    final_chain = random_final_chain(left_chains, right_chains)
+    return final_chain if final_chain
+    extended_chains_by_one_degree!(right_chains)
+    final_chain = random_final_chain(left_chains, right_chains)
+    return final_chain if final_chain
+    find_chain(left_chains, right_chains)
+  end
+
+  def self.extended_chains_by_one_degree!(chains)
+    chains.map! do |chain|
+      Linking.find(chain.last).wca_ids.map { |wca_id| [*chain, wca_id] }
```

Comment:
As far as I can tell, there's no logic here that prevents revisiting a wca_id that you've already visited. Will this cause an infinite loop in the event that there is no chain between two people? Could you add a test for that to the `ComputeRelationsAuxiliaryData` spec I asked you to create?